tools/wpt/run.py

```diff
@@ -487,6 +489,19 @@ def setup_wptrunner(venv, prompt=True, install_browser=False, **kwargs):
     setup_cls = product_setup[kwargs["product"]](venv, prompt, sub_product)
     setup_cls.install_requirements()
 
+    affected_revish = kwargs.pop("affected", None)
+    if affected_revish is not None:
+        files_changed, _ = testfiles.files_changed(affected_revish, include_uncommitted=True, include_new=True)
+        # TODO: honor --no-manifest-update and other manifest args
+        tests_changed, tests_affected = testfiles.affected_testfiles(files_changed)
+        test_list = tests_changed | tests_affected
+        if not test_list:
+            # TODO: make something show up in all loggers, and also still write wpt_report.json?
```

Comment:
@jgraham this is the main outstanding part that I need advice on. This code with the default logger doesn't print anything or create a wpt_report.json if --log-wptreport is used. Where is the code that decides to run all tests if none are given? I think there, is `--affected` is used, would be the place to terminate in a way that achieves what the TODO asks for.

And how do I make the default logger print things?

Reply:
`--log-mach -` should make us get the default output again, at least

Reply:
That's fine for CI, but when people run `./wpt run --affected HEAD^1` or similar locally, I'd like there to be clear if no tests are run, instead of silently exiting.

Reply:
I think this is a generic problem with the default logger. I think the right solution is to allow log messages until a flag is set in the logger and then turn them off once the run starts (since the point of the logger is to be as non-verbose as possible). I don't know if it's worthwhile to do something special here instead of just fixing the generic problem.